File: C#/Huffman Coding/Node.cs
namespace Huffman_Coding
{
    public class Node
    {
        public int left, right, root;

        public Node leftNode, rightNode, parent;

        public Node(int left, int right)
        {
            this.left = left;
            this.right = right;
            this.root = left + right;
        }

        public override string ToString()
        {
            return "Root = " + root + ", Left = " + left + ", Right = " + right;
        }
    }
}

File: C/node.c
#include "node.h"

static const char *FORMAT = "Root = %d, Left = %d, Right = %d\n";      // a buffer to be returned...

void initializeNode(int left, int right, struct Node *node)
{
    (*node).left = left;
    (*node).right = right;
    (*node).root = left + right;
    (*node).leftNode = (*node).rightNode = (*node).parent = NULL;
}

void printNode(struct Node node)
{
    printf(FORMAT, node.root, node.left, node.right);
}
